### PR TITLE
Prodigal

### PR DESCRIPTION
**Pull request by zhaoc1**
* [ ] I have run `bash tests/test.sh` on a local deployment and the tests passed successfully
The tests/test.sh intentionally failed, for the purpose of fixing the bugs in the aggregate_results (details mentioned in the following).
* [ ] If this adds a new output file, I have added a check to tests/targets.txt
* [x] If this fixes a bug, I have added an appropriate test to tests/test.sh
* [ ] If this adds or modifies a rule that uses FASTQ files, the input accepts gzipped FASTQ and outputs gzipped FASTQ, or marks uncompressed FASTQ output as `temp`

This branch updated the orf finding using prodigal, and update following up blast.rules.

TODO: `cap3_contig1` may have more than one ORFs from prodigal result: `cap_contig1_1`, `cap_contig1_2" and etc. Right now, the `aggregate_results` rule can't merge blastpx report (ORFs as input) with the blastn report (contigs as input). And the counts in the summary/{sample}.tsv is always 0. Need to fix this parsing problem.

### CODE REVIEW

File: rules/annotation/orf.rules [hunk 1]
```diff
@@ -1,45 +1,41 @@
...
-        str(ANNOTATION_FP/'genes'/'mga'/'{sample}_contigs.genes')
+        gff = str(ANNOTATION_FP/'genes'/'prodigal'/'{sample}_genes.gff'),
+        faa = str(ANNOTATION_FP/'genes'/'prodigal'/'{sample}_genes_prot.fa'),
+        fna = str(ANNOTATION_FP/'genes'/'prodigal'/'{sample}_genes_nucl.fa'),
+        log = str(ANNOTATION_FP/'genes'/'prodigal'/'log'/'{sample}.out')
     shell:
-        """mga -m {input} > {output}"""
+        """
+        if [[ -s {input} ]]; then
+          echo "yes"
```
Comment (eclarke) [inline rules/annotation/orf.rules]: Let's remove or make more verbose these debug messages (i.e. "No contigs; skipping ORF finding")

**Comment by eclarke** [comment]
Can you make a (currently failing) test for the bug you mentioned in the commit? I can work on fixing it since I know you're busy. I'd rather not merge this until that's been fixed though because it sounds like it technically would break the report function.

File: rules/annotation/orf.rules (hunk 1 above)
Comment (zhaoc1) [reply rules/annotation/orf.rules]: My bad. that's only my own testing purpose. I will remove them right away.

**Comment by zhaoc1** [comment]
Now, the unit test failed at checking the card counts in the `summary.tsv`. The problem is the inconsistent naming between contig-level for blastn and orf-level for blastpx (one-to-multiple mapping between contig and orfs).

**Pull request merged**